### ISSUE TITLE
Success Condition(s): done[0] is not equal to info["won"][0]

### ISSUE THREAD

**Issue by ai-nikolai** [alfworld/alfworld#51]
In running various evaluations, it happened that the agent would receive a `done==True` signal, however, `info["won"][0]==False`.

Looking at the traces the agent was not successful.

This raises the question of what and how to check for success conditions properly, as many works use `done==True` to check for success. 

Please elaborate. Thank you @MarcCote @MohitShridhar 

---
_On a related note:_
Our code does not seem to explicitly pass config parameters into the environment, however, `done==True` was returned after 50 steps, the "Alfworld" code indicates `max_steps=2000`, however.

**Comment by MarcCote** [comment]
@ai-nikolai the `max_nb_steps_per_episode` is indeed 50 which will cause `done=True`.
https://github.com/alfworld/alfworld/blob/master/configs/base_config.yaml#L88

`info["won"]` is set to True only when the task is completed successfully.

The 2000 you seem to be referring to is for the non-text version of ALFRED.

**Comment by ai-nikolai** [comment]
Perfect. Thank you very much. This is very helpful.

Therefore for works that use `done=True` to measure success it meant that their results are actually not correct (or very likely not correct), right?

**Comment by MarcCote** [comment]
I agree with you. Measuring accuracy based on `done==True` is incorrect.

ALFWorld follows the Openai Gym's convention for `done` (see https://gymnasium.farama.org/api/env/#gymnasium.Env.step). It has now been deprecated in favor of two return values: `terminated` (akin to `infos['won']`) and `truncated` (i.e., `max_steps_per_episode` was reached) which have the benefit of being clearer.

**Comment by ai-nikolai** [comment]
Thanks for this clarification and referencing the newer interface.

**Comment by MarcCote** [comment]
@ai-nikolai when you say some works use `done` to measure success, do they also look at the score/reward? If so, that would be equivalent to checking `infos['won']` since ALFWorld is a sparse reward type of environment.

**Comment by ai-nikolai** [comment]
I need to investigate further, but what I saw at the moment is that they use `done=True` only as the success metric (and therefore obviously get great results (that are not reproducible).
